ekwipunek.py

```diff
@@ -0,0 +1,823 @@
+    :param magazyn: słownik zawierający wszystkie kontenery przypisane
+        do odpowiedniego klucza
+    :param w_uzyciu: kontener zawierający przedmioty znajdujące się w danej
+        chwili w użyciu
+    """
+    def __init__(self):
+        self._bronie = Bronie()
+        self._pancerze = Pancerze()
+        self._przedmioty_magiczne = PrzedmiotyMagiczne()
+        self._pisma = Pisma()
+        self._artefakty = Artefakty()
+        self._jedzenie = Zywnosc()
+        self._pozostale = Pozostalosci()
+        self._magazyn = {
+            "Bronie": self._bronie,
+            "Pancerze": self._pancerze,
+            "Przemioty Magiczne": self._przedmioty_magiczne,
+            "Pisma": self._pisma,
+            "Artefakty": self._artefakty,
+            "Jedzenie": self._jedzenie,
+            "Pozostałe": self._pozostale,
+        }
+        self._w_uzyciu = {}
+
+    @property
+    def bronie(self):
+        return self._bronie
+
+    @property
+    def pancerze(self):
+        return self._pancerze
+
+    @property
+    def przedmioty_magiczne(self):
+        return self._przedmioty_magiczne
+
+    @property
+    def pisma(self):
+        return self._pisma
+
+    @property
+    def artefakty(self):
+        return self._artefakty
+
+    @property
+    def jedzenie(self):
+        return self._jedzenie
+
+    @property
+    def pozostale(self):
+        return self._pozostale
+
+    @property
+    def w_uzyciu(self):
+        return self._w_uzyciu
+
+    @property
+    def magazyn(self):
+        return self._magazyn
+
+    def wyrzuc(self):
+        """
+        Metoda wyrzuć wyrzucająca dany przedmiot z obiektu klasy Ekwipunek i zwracająca go
+        do obiektu klasy Podłoga.
+        """
+        print("Przedmioty, które możesz wyrzucić, lub wpisz 'wyjscie', by wyjść:")
+        for key, value in self.wyswietl_magazyn().items():
+            for k, v in value.items():
+                print(k + " sztuk " + str(len(v)))
+        val = input("Podaj nazwę przedmiotu, który chcesz wyrzucić")
+        if val == "wyjscie":
+            pass
+        else:
+            for key, value in self.wyswietl_magazyn().items():
+                if val in value:
+                    przedmiot = self.magazyn[key][val][0]
+                    Podloga_Obiekt.zawartosc.append(przedmiot)
+                    # mogę wyrzucić przedmiot w uzyciu, ale wtedy wylatuje on zarowno z w_uzyciu jak
+                    # i z kontenera (ZAWSZE), wynika to z linijki
+                    # przedmiot = self.magazyn[key][val][0]
+                    # poniewaz taka sama znajduje sie w metodzie uzyj (oraz Oselka)
+                    for k, v in self.w_uzyciu.copy().items():
+                        if przedmiot == v:
+                            del self.w_uzyciu[k]
+                    if isinstance(przedmiot, Bron):
+                        self._bronie[przedmiot.nazwa].remove(przedmiot)
+                        if len(self._bronie[przedmiot.nazwa]) == 0:
```

Comment:
ten check jest niepotrzebny bo skoro ilość jest 0 to w ogóle nie powinien widnieć w magazynie

Reply:
Nie no, jest potrzebne -> skoro wewnątrz magazynu nazwa przedmiotu jest kluczem do danego obiektu, to jeśli nie będziemy tego sprawdzać to, dla przykładu wyrzucając przedmiot burzy będziemy mieć klucz przedmiot burzy ale żaden obiekt nie będzie do niego przypisany

Reply:
![Screenshot 2024-03-05 at 13 35 31](https://github.com/szkudlarekcmd/ekwipunek/assets/84017946/23cfcefd-9f3a-43fc-8dba-fc7f728cf3b1)